src/relayclient/RelayProvider.ts

```diff
@@ -25,39 +26,36 @@ export type JsonRpcCallback = (error: Error | null, result?: JsonRpcResponse) =>
 interface ISendAsync {
   sendAsync?: any
 }
-
-export class RelayProvider implements HttpProvider {
+// TODO: stop faking the HttpProvider implementation -  it won't work for any other 'origProvider' type
+export class RelayProvider implements HttpProvider, ISendProvider {
   protected readonly origProvider: HttpProvider & ISendAsync
   private readonly origProviderSend: any
   protected config!: GSNConfig
 
   readonly relayClient: RelayClient
   logger!: LoggerInterface
 
+  static newProvider (input: GSNUnresolvedConstructorInput): RelayProvider {
+    return new RelayProvider(new RelayClient(input))
+  }
+
   /**
    * create a proxy provider, to relay transaction
-   * @param overrideDependencies
-   * @param relayClient
-   * @param origProvider - the underlying web3 provider
-   * @param gsnConfig
    */
-  constructor (origProvider: HttpProvider, gsnConfig: Partial<GSNConfig>, overrideDependencies?: Partial<GSNDependencies>, relayClient?: RelayClient) {
-    this.host = origProvider.host
-    this.connected = origProvider.connected
+  constructor (
+    relayClient: RelayClient
+  ) {
+    this.relayClient = relayClient
+    this.origProvider = this.relayClient.getUnderlyingProvider()
+    this.host = this.origProvider.host
+    this.connected = this.origProvider.connected
```

Comment:
wow. this "connected" is buggy, since we copy the current connected status, and save it.

Reply:
It was like that, and unfortunately not much we can do here

Reply:
created https://opengsn.atlassian.net/browse/OG-326